### ISSUE TITLE
How to get a builded engine weight？

### ISSUE THREAD

**Issue by Galaxygray** [NVIDIA/TensorRT#2545]
## Description
I want to get a builded engine weight，but I‘m not find any get weight API，but only some get weight description API.
How to get a builded engine weight？

## Environment

Description
There is two thread doing inference in front and one backup thread loading trt engine, but when the backup thread was loading myelingraph, Doing inference thread had been locked 300ms+, In normal the inference thread cost 1-2ms do once inference including memcpy.
Is there have some bug?
image

Environment
TensorRT Version: 8.2.2.1
NVIDIA GPU: T4
NVIDIA Driver Version: 450.102.04
CUDA Version: 10.2
CUDNN Version: 8.2.4
Operating System: centos7.2
Python Version (if applicable): 3.8
Tensorflow Version (if applicable): No
PyTorch Version (if applicable): No

**Comment by zerollzeng** [comment]
I don't think we support that, @nvpohanh correct me if I'm wrong

**Comment by nvpohanh** [comment]
> I want to get a builded engine weight，but I‘m not find any get weight API，but only some get weight description API.
How to get a builded engine weight？

There is no API to get that.

**Comment by Galaxygray** [comment]
OK,Thanks very much, Is there has support plan？

**Comment by nvpohanh** [comment]
Not on the roadmap.

This is not a trivial change because TRT does aggressive fusions and layout transformations, so the weights may have been fused with the layer parameters with other layers and the layout may also have been shuffled for the best memory loading pattern. This makes it hard to retrieve the original weights from TRT engine.

**Comment by Galaxygray** [comment]
OK, Thanks！
